### PR TITLE
VIH-7082 fix for ithc to get logged user

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7082
### Change description ###
Fix to get Logged user in ngOnInit

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/vho-chat/vho-chat.component.ts [hunk 1]
```diff
@@ -80,8 +81,9 @@ export class VhoChatComponent extends ChatBaseComponent implements OnInit, OnDes
     async ngOnInit() {
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/vho-chat/vho-chat.component.ts:81]: please remove the `async` on `ngOnInit`

**Review by shaed-parkar — CHANGES_REQUESTED** [review]
Since we need data before components load, I would highly recommend you introduce a ResolveGuard to get the current participant for the given conference id before the component loads

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/vho-chat/vho-chat.component.ts (hunk 1 above)
Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/vho-chat/vho-chat.component.ts]: Thanks, Shaed! done

Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/vho-chat/vho-chat.component.ts]: Implemented Resolve Guard.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/tests/individual-participant-status-list.component.availability.spec.ts [hunk 2]
```diff
@@ -48,20 +50,28 @@ describe('IndividualParticipantStatusListComponent Participant Status and Availa
...
-                    role: Role.Judge
-                })
-            )
-        );
+        logged = new LoggedParticipantResponse({
+            participant_id: conference.participants[2].id,
+            display_name: 'Jonh Doe',
+            role: Role.Judge
+        });
+        videoWebService.getCurrentParticipant.and.returnValue(Promise.resolve(logged));
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/tests/individual-participant-status-list.component.availability.spec.ts:58]: Does the spec still need `getCurrentParticipant` to be set up anymore now that the component gets the data from via the resolve guard now?

Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/tests/individual-participant-status-list.component.availability.spec.ts]: Removed

**Review by shaed-parkar — APPROVED** [review]
Nice work!

**Pull request merged**